tests/simpleharness/GetAllocatedBytesForCurrentThreadTest.cs

```diff
@@ -0,0 +1,64 @@
+﻿using Microsoft.Xunit.Performance;
+using System;
+using System.Reflection;
+
+namespace simpleharness
+{
+    public class GetAllocatedBytesForCurrentThreadTest
+    {
+        [Benchmark]
+        public static void WithCollect()
+        {
+            var method = typeof(GC)
+                .GetTypeInfo()
```

Comment:
I think it would be good to have a project of functional tests that we run as well that run as part of the build.  These should live there.  It's good that they are [Benchmark]s and not [Fact]s, because we'll test our code, but I think that simpleharness should just be a simple example, and we should put all of these great test cases that you're building into a functional tests project that we run on every build.  Then devs will see breaks from their code, and the CI will automatically run them as well.